keras/callbacks.py

```diff
@@ -428,6 +429,48 @@ def on_epoch_end(self, epoch, logs=None):
                     self.model.save(filepath, overwrite=True)
 
 
+class DeadReluDetector(Callback):
+    def __init__(self, original_model, x_train):
+        if not isinstance(original_model, Sequential):
```

Comment:
I end up relying on `Sequental` anyway
The reason is that I have to reconstruct truncated models to check whether or not neurons fire, and I have no idea how to reconstruct anything but `Sequental`